drivers/spi/spi_gecko_eusart.c

```diff
@@ -0,0 +1,325 @@
+		UNALIGNED_PUT(rx_frame, (uint8_t *)data->ctx.rx_buf);
+	}
+	spi_context_update_rx(&data->ctx, 1, 1);
+	return 0;
+}
+
+static void spi_gecko_eusart_xfer(const struct device *dev, const struct spi_config *config)
+{
+	int ret;
+	struct spi_gecko_eusart_data *data = dev->data;
+	struct spi_context *ctx = &data->ctx;
+	const struct spi_gecko_eusart_config *gecko_config = dev->config;
+
+	spi_context_cs_control(ctx, true);
+
+	do {
+		ret = spi_eusart_shift_frames(gecko_config->base, data);
+	} while (!ret && spi_eusart_transfer_ongoing(data));
+
+	spi_context_cs_control(ctx, false);
+	spi_context_complete(ctx, dev, 0);
+}
+
+/* API Functions */
+static int spi_gecko_eusart_init(const struct device *dev)
+{
+	int err;
+	const struct spi_gecko_eusart_config *config = dev->config;
+	struct spi_gecko_eusart_data *data = dev->data;
+
+	err = pinctrl_apply_state(config->pcfg, PINCTRL_STATE_DEFAULT);
+	if (err < 0) {
+		return err;
+	}
+
+	err = spi_context_cs_configure_all(&data->ctx);
+	if (err < 0) {
+		return err;
+	}
+	spi_context_unlock_unconditionally(&data->ctx);
+
+	return 0;
+}
+
+static int spi_gecko_eusart_transceive(const struct device *dev, const struct spi_config *config,
+				       const struct spi_buf_set *tx_bufs,
+				       const struct spi_buf_set *rx_bufs)
+{
+	struct spi_gecko_eusart_data *data = dev->data;
+	uint16_t control = 0;
+	int ret;
+
+	spi_context_lock(&data->ctx, false, NULL, NULL, config);
+
+	ret = spi_eusart_config(dev, config, &control);
+	if (ret < 0) {
+		spi_context_release(&data->ctx, ret);
+		return ret;
+	}
+
+	spi_context_buffers_setup(&data->ctx, tx_bufs, rx_bufs, 1);
+	spi_gecko_eusart_xfer(dev, config);
+
+	spi_context_release(&data->ctx, ret);
+
+	return 0;
+}
+
+#ifdef CONFIG_SPI_ASYNC
+static int spi_gecko_eusart_transceive_async(const struct device *dev,
+					     const struct spi_config *config,
+					     const struct spi_buf_set *tx_bufs,
+					     const struct spi_buf_set *rx_bufs,
+					     struct k_poll_signal *async)
+{
+	return -ENOTSUP;
+}
+#endif /* CONFIG_SPI_ASYNC */
+
+static int spi_gecko_eusart_release(const struct device *dev, const struct spi_config *config)
+{
+	const struct spi_gecko_eusart_config *gecko_config = dev->config;
+	struct spi_gecko_eusart_data *data = dev->data;
+
+	spi_context_unlock_unconditionally(&data->ctx);
+
+	if (!(gecko_config->base->STATUS & EUSART_STATUS_TXIDLE)) {
+		return -EBUSY;
+	}
+	return 0;
+}
+
+/* Device Instantiation */
+static const struct spi_driver_api spi_gecko_eusart_api = {
+	.transceive = spi_gecko_eusart_transceive,
+#ifdef CONFIG_SPI_ASYNC
+	.transceive_async = spi_gecko_eusart_transceive_async,
+#endif /* CONFIG_SPI_ASYNC */
+	.release = spi_gecko_eusart_release,
+};
+
+#define SPI_INIT(n) \
+	PINCTRL_DT_INST_DEFINE(n); \
+	static struct spi_gecko_eusart_data spi_gecko_eusart_data_##n = { \
+		SPI_CONTEXT_INIT_LOCK(spi_gecko_eusart_data_##n, ctx), \
+		SPI_CONTEXT_INIT_SYNC(spi_gecko_eusart_data_##n, ctx), \
```

Comment:
please, format this properly (`\` to the same column), clang-format can help

Reply:
Done